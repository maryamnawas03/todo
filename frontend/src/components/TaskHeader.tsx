// src/components/TaskHeader.tsx
import React from "react";
import { CheckSquare } from "lucide-react";

export const TaskHeader: React.FC = () => {
  return (
    <div className="text-center mb-8">
      <div className="flex justify-center mb-4">
        <div className="w-14 h-14 bg-gradient-to-br from-purple-500 to-purple-600 rounded-2xl flex items-center justify-center shadow-lg">
          <CheckSquare className="w-8 h-8 text-white" strokeWidth={2.5} />
        </div>
      </div>

      <h1 className="text-4xl font-bold bg-gradient-to-r from-purple-600 to-indigo-600 bg-clip-text text-transparent mb-2">
        TaskFlow
      </h1>

      <p className="text-gray-600 text-lg">
        Master your productivity, one task at a time
      </p>
    </div>
  );
};
