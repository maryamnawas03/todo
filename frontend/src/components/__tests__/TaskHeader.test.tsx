// src/components/__tests__/TaskHeader.test.tsx
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { TaskHeader } from '../TaskHeader';

describe('TaskHeader', () => {
  describe('Rendering', () => {
    it('should render the app title "TaskFlow"', () => {
      render(<TaskHeader />);
      
      expect(screen.getByText('TaskFlow')).toBeInTheDocument();
    });

    it('should render the subtitle', () => {
      render(<TaskHeader />);
      
      expect(screen.getByText('Master your productivity, one task at a time')).toBeInTheDocument();
    });

    it('should display both title and subtitle', () => {
      render(<TaskHeader />);
      
      expect(screen.getByText('TaskFlow')).toBeInTheDocument();
      expect(screen.getByText(/master your productivity/i)).toBeInTheDocument();
    });

    it('should render heading as h1', () => {
      render(<TaskHeader />);
      
      const heading = screen.getByRole('heading', { level: 1 });
      expect(heading).toHaveTextContent('TaskFlow');
    });
  });

  describe('Content', () => {
    it('should have centered layout', () => {
      render(<TaskHeader />);
      
      const title = screen.getByText('TaskFlow');
      const subtitle = screen.getByText(/master your productivity/i);
      
      expect(title).toBeInTheDocument();
      expect(subtitle).toBeInTheDocument();
    });
  });

  describe('Snapshot', () => {
    it('should match snapshot', () => {
      const { asFragment } = render(<TaskHeader />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
