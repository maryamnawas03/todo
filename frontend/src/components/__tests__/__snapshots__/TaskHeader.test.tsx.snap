// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskHeader Snapshot should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center mb-8"
  >
    <div
      class="flex justify-center mb-4"
    >
      <div
        class="w-14 h-14 bg-gradient-to-br from-purple-500 to-purple-600 rounded-2xl flex items-center justify-center shadow-lg"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-square-check-big w-8 h-8 text-white"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2.5"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 10.656V19a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h12.344"
          />
          <path
            d="m9 11 3 3L22 4"
          />
        </svg>
      </div>
    </div>
    <h1
      class="text-4xl font-bold bg-gradient-to-r from-purple-600 to-indigo-600 bg-clip-text text-transparent mb-2"
    >
      TaskFlow
    </h1>
    <p
      class="text-gray-600 text-lg"
    >
      Master your productivity, one task at a time
    </p>
  </div>
</DocumentFragment>
`;
